{"uid":"735b3ec391f2066e","name":"test_persist_data_through_primary_deletion","fullName":"tests.integration.test_charm#test_persist_data_through_primary_deletion","historyId":"9d1cbaaaf2bdf66102409771e5d14ec7","time":{"start":1723258102844,"stop":1723259605372,"duration":1502528},"description":"Test data persists through a primary deletion.","descriptionHtml":"<p>Test data persists through a primary deletion.</p>\n","status":"broken","statusMessage":"asyncio.exceptions.TimeoutError: Timed out waiting for model:\npostgresql/4 [executing] active:","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: TResult | None = func()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 242, in <lambda>\n    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/threadexception.py\", line 92, in pytest_runtest_call\n    yield from thread_exception_runtest_hook()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/threadexception.py\", line 68, in thread_exception_runtest_hook\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/unraisableexception.py\", line 95, in pytest_runtest_call\n    yield from unraisable_exception_runtest_hook()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/unraisableexception.py\", line 70, in unraisable_exception_runtest_hook\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/logging.py\", line 848, in pytest_runtest_call\n    yield from self._runtest_for(item, \"call\")\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/logging.py\", line 831, in _runtest_for\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/capture.py\", line 879, in pytest_runtest_call\n    return (yield)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/skipping.py\", line 257, in pytest_runtest_call\n    return (yield)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 174, in pytest_runtest_call\n    item.runtest()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1627, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 159, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 529, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/tests/integration/test_charm.py\", line 344, in test_persist_data_through_primary_deletion\n    await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status=\"active\", timeout=1500)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 2969, in wait_for_idle\n    raise jasyncio.TimeoutError(\"Timed out waiting for model:\\n\" + busy)\nasyncio.exceptions.TimeoutError: Timed out waiting for model:\npostgresql/4 [executing] active:","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":1,"retriesStatusChange":true,"beforeStages":[{"name":"tmp_path_factory","time":{"start":1723254363912,"stop":1723254363912,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop","time":{"start":1723254363912,"stop":1723254363912,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test","time":{"start":1723254363912,"stop":1723254364072,"duration":160},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail","time":{"start":1723258102844,"stop":1723258102844,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Test data persists through a primary deletion.","status":"broken","statusMessage":"asyncio.exceptions.TimeoutError: Timed out waiting for model:\npostgresql/4 [executing] active:","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: TResult | None = func()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 242, in <lambda>\n    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/threadexception.py\", line 92, in pytest_runtest_call\n    yield from thread_exception_runtest_hook()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/threadexception.py\", line 68, in thread_exception_runtest_hook\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/unraisableexception.py\", line 95, in pytest_runtest_call\n    yield from unraisable_exception_runtest_hook()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/unraisableexception.py\", line 70, in unraisable_exception_runtest_hook\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/logging.py\", line 848, in pytest_runtest_call\n    yield from self._runtest_for(item, \"call\")\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/logging.py\", line 831, in _runtest_for\n    yield\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/capture.py\", line 879, in pytest_runtest_call\n    return (yield)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 167, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/skipping.py\", line 257, in pytest_runtest_call\n    return (yield)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 174, in pytest_runtest_call\n    item.runtest()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1627, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 159, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 529, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/tests/integration/test_charm.py\", line 344, in test_persist_data_through_primary_deletion\n    await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status=\"active\", timeout=1500)\n  File \"/home/runner/work/postgresql-operator/postgresql-operator/.tox/integration/lib/python3.10/site-packages/juju/model.py\", line 2969, in wait_for_idle\n    raise jasyncio.TimeoutError(\"Timed out waiting for model:\\n\" + busy)\nasyncio.exceptions.TimeoutError: Timed out waiting for model:\npostgresql/4 [executing] active:","steps":[],"attachments":[{"uid":"402919de56256ca1","name":"log","source":"402919de56256ca1.txt","type":"text/plain","size":9318}],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"ops_test::finalizer","time":{"start":1723259605373,"stop":1723259605759,"duration":386},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail::0","time":{"start":1723259605373,"stop":1723259605373,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::2","time":{"start":1723259605760,"stop":1723259605760,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_close_event_loop","time":{"start":1723259605760,"stop":1723259605760,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_provide_clean_event_loop","time":{"start":1723259605760,"stop":1723259605760,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"group(1)"},{"name":"tag","value":"asyncio"},{"name":"parentSuite","value":"tests.integration"},{"name":"suite","value":"test_charm"},{"name":"host","value":"fv-az842-614"},{"name":"thread","value":"8729-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.integration.test_charm"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"f496733c4d00d0bf","status":"unknown","time":{}}],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":28,"skipped":1,"passed":149,"unknown":3,"total":181},"items":[{"uid":"36b2a97e1d53b803","reportUrl":"../3094//#testresult/36b2a97e1d53b803","status":"passed","time":{"start":1723169762911,"stop":1723170080469,"duration":317558}},{"uid":"529a352621eb8016","reportUrl":"../3091//#testresult/529a352621eb8016","status":"passed","time":{"start":1723083711705,"stop":1723084016442,"duration":304737}},{"uid":"bdfd622f730330","reportUrl":"../3089//#testresult/bdfd622f730330","status":"passed","time":{"start":1722996572051,"stop":1722996876885,"duration":304834}},{"uid":"2b597753f6b6f917","reportUrl":"../3086//#testresult/2b597753f6b6f917","status":"passed","time":{"start":1722910595569,"stop":1722910930677,"duration":335108}},{"uid":"e7fdd496988195ed","reportUrl":"../3081//#testresult/e7fdd496988195ed","status":"passed","time":{"start":1722823948321,"stop":1722824229509,"duration":281188}},{"uid":"b39aec3af4fc0b08","reportUrl":"../3080//#testresult/b39aec3af4fc0b08","status":"passed","time":{"start":1722737301551,"stop":1722737596951,"duration":295400}},{"uid":"4f8531ce11f22fb8","reportUrl":"../3070//#testresult/4f8531ce11f22fb8","status":"passed","time":{"start":1722565148100,"stop":1722565459881,"duration":311781}},{"uid":"6aff253e9ab671be","reportUrl":"../3067//#testresult/6aff253e9ab671be","status":"passed","time":{"start":1722478028455,"stop":1722478296962,"duration":268507}},{"uid":"cd8ce39a673fc0b1","reportUrl":"../3056//#testresult/cd8ce39a673fc0b1","status":"passed","time":{"start":1722391595415,"stop":1722391889554,"duration":294139}},{"uid":"f07fe3fca0b8cdc5","reportUrl":"../3034//#testresult/f07fe3fca0b8cdc5","status":"passed","time":{"start":1722304963147,"stop":1722305299870,"duration":336723}},{"uid":"18b91312bfb3b183","reportUrl":"../3016//#testresult/18b91312bfb3b183","status":"passed","time":{"start":1722218919404,"stop":1722219259534,"duration":340130}},{"uid":"c051d55d2962e42a","reportUrl":"../3009//#testresult/c051d55d2962e42a","status":"passed","time":{"start":1722132524317,"stop":1722132849436,"duration":325119}},{"uid":"323602f947157605","reportUrl":"../3004//#testresult/323602f947157605","status":"passed","time":{"start":1722045749055,"stop":1722046057461,"duration":308406}},{"uid":"25e41f099e18a190","reportUrl":"../3001//#testresult/25e41f099e18a190","status":"passed","time":{"start":1721959660513,"stop":1721959987221,"duration":326708}},{"uid":"5d078bdaa22444d1","reportUrl":"../2992//#testresult/5d078bdaa22444d1","status":"passed","time":{"start":1721872932064,"stop":1721873251977,"duration":319913}},{"uid":"15a18d28db61e8f","reportUrl":"../2979//#testresult/15a18d28db61e8f","status":"passed","time":{"start":1721787599821,"stop":1721787881730,"duration":281909}},{"uid":"2eb6cbf44cc75ef2","reportUrl":"../2975//#testresult/2eb6cbf44cc75ef2","status":"passed","time":{"start":1721700670436,"stop":1721700959804,"duration":289368}},{"uid":"bc46a47a0d27a72c","reportUrl":"../2969//#testresult/bc46a47a0d27a72c","status":"passed","time":{"start":1721614129966,"stop":1721614447319,"duration":317353}},{"uid":"9870a9b42d365549","reportUrl":"../2968//#testresult/9870a9b42d365549","status":"passed","time":{"start":1721528350018,"stop":1721528661059,"duration":311041}},{"uid":"32805802f12b43fa","reportUrl":"../2967//#testresult/32805802f12b43fa","status":"passed","time":{"start":1721441378812,"stop":1721441676329,"duration":297517}}]},"tags":["asyncio","group(1)"]},"source":"735b3ec391f2066e.json","parameterValues":[]}